lib/mix/test/mix/tasks/test_test.exs

```diff
@@ -257,6 +258,75 @@ defmodule Mix.Tasks.TestTest do
     end)
   end
 
+  describe "--max-failures" do
```

Comment:
I don't believe we need those integration tests as they are very expensive. We should be able to check all of those cases through the ExUnit suite itself.